### PR TITLE
fix toolchain build/push scripts

### PR DESCRIPTION
**Pull request by filariow**
Changes to the CI have been tested [here](https://github.com/filariow/workspaces/actions/runs/8740686163/job/23984932310?pr=4)

### CODE REVIEW

File: hack/demo.sh [hunk 1]
```diff
@@ -11,9 +11,25 @@ SCRIPT_DIR="$(dirname "${CURRENT_DIR}")"
 SUFFIX="e2e$(date +'%d%H%M%S')"
 echo "using suffix: ${SUFFIX}"
 
-# build and install toolchain
-( "${SCRIPT_DIR}/toolchain_build_push_member.sh" "${SUFFIX}" )
-( "${SCRIPT_DIR}/toolchain_build_push_host.sh" "${SUFFIX}" )
+( 
+  # create a temporary direction
```
Comment (filariow) [inline hack/demo.sh:15]:
```suggestion
  # create a temporary directory
```

File: .github/workflows/push.yaml [hunk 2]
```diff
@@ -46,14 +49,20 @@ jobs:
           go-version: ${{ env.GO_VERSION }}
           cache: false
 
+      - name: Compute base image
+        id: img
+        run: |
+          IMG="quay.io/${{ env.QUAY_NAMESPACE }}/${{ matrix.image_base }}"
+          echo "base=${IMG}" >> ${GITHUB_OUTPUT}
+
```
Comment (sadlerap) [inline .github/workflows/push.yaml:57]: The stuff under the `build-workspaces` job doesn't touch anything related to the kubesaw toolchain.  I'm trying to understand what these changes are supposed to accomplish.  What do they add that the old version of this job didn't do?

I'm fine with this change, just puzzled about its purpose.

**Review by sadlerap — COMMENTED** [review]
largely lgtm, just a question

File: .github/workflows/push.yaml (hunk 2 above)
Comment (filariow) [reply .github/workflows/push.yaml:57]: They makes the life way easier when testing the changes to ci in a fork

Comment (filariow) [reply .github/workflows/push.yaml:57]: We can move them to another PR if we prefer, I forgot them here

**Comment by sadlerap** [comment]
@filariow Do we want to continue this PR in light of https://github.com/codeready-toolchain/toolchain-e2e/pull/951?

**Comment by sadlerap** [comment]
@filariow I'm not sure this PR makes sense anymore, not after #94.  If this is something you want to pursue further, I'd recommend splitting out the changes for building & pushing `server` and `operator` from the other changes.  Those still seem useful, while the changes to how we build kubesaw components don't anymore.

**Comment by filariow** [comment]
Closing as superseded by https://github.com/konflux-workspaces/workspaces/pull/94 and https://github.com/konflux-workspaces/workspaces/pull/110

**Pull request closed without merging**